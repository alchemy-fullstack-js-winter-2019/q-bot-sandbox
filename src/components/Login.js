
import React, { Component } from 'react';
import { subscribe } from '../services/firebase';

export default class Login extends Component {
  state = {
    provider: '',
    loading: false
  }

  componentDidUpdate() {
    subscribe(user => {
      console.log(user)
    }, this.state.provider)
  }

  componentDidMount() {
    subscribe(user => {
      if(user) {
        this.setState({ loading: true })
        this.props.history.replace('/questions');
      }
    })
  }

  handleClick = ({ target }) => {
    this.setState({ provider: target.value })
  }

  render() {
    return (
      <>
        {this.state.loading === false && <><h1>Q BOT LOGIN</h1>
        <button name='google' value='google' onClick={this.handleClick}>Google</button>
        <button name='github' value='github' onClick={this.handleClick}>GitHub</button></>}
      </>
    );
  }
 
}
