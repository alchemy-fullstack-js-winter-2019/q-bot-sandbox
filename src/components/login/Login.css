@import url('https://fonts.googleapis.com/css?family=Roboto:400,700');

main {
  text-align: center;
  font-family: monserrat;
  margin-top: 6em;
}

.login-btn {
  font-family: 'Roboto';
  border: none;
  border-radius: .51em;
  height: 2.3em;
  background-color: #aaaaaa;
  color: white;
  font-weight: 400;
  font-size: 1.3rem;
  cursor: pointer;
  letter-spacing: .1rem;
  padding: .4em;
  margin: 1em;
}

.login-btn:hover {
  background-color: #929292;
}

.logo-div {
  display: flex;
  justify-content: center;
  align-items: center;
}

.logo-div:after {
  clear:both;
  display:block;
  content:'';
}

.logo-div img:hover {
  -webkit-animation: shake .5s ease-in-out;
  animation: shake .5s ease-in-out;
}
@-webkit-keyframes shake {
  0% {
    -webkit-transform: translateX(0);
  }
  20% {
    -webkit-transform: translateX(-.5em);
  }
  40% {
    -webkit-transform: translateX(.2em);
  }
  60% {
    -webkit-transform: translateX(-.5em);
  }
  80% {
    -webkit-transform: translateX(.2em);
  }
  100% {
    -webkit-transform: translateX(0);
  }
}
@keyframes shake {
  0% {
    transform: translateX(0);
  }
  20% {
    transform: translateX(-.5em);
  }
  40% {
    transform: translateX(.2em);
  }
  60% {
    transform: translateX(-.5em);
  }
  80% {
    transform: translateX(.2em);
  }
  100% {
    transform: translateX(0);
  }
}

.logo-div h1 {
  font-family: 'Roboto Mono', monospace;
  font-size: 4em;
  margin: .4em;
}

.logo-div img {
  width: 4em;
  height: 4em;
}

@media screen and (min-width: 320px) and (max-width: 400px) {
  .logo-div {
    margin-top: 2em;
    display: block;
  }

  .logo-div h1 {
    margin: 0;
  }
}
