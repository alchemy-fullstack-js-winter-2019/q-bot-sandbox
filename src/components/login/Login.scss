//colors
$color_silver_chalice: #aaaaaa;
$white: white;
$color_mountain_mist: #929292;

//imports
@import url('https://fonts.googleapis.com/css?family=Roboto:400,700');

//fonts
$font_0: monserrat;
$font_1: Roboto;
$font_3: monospace;

main {
	text-align: center;
	font-family: $font_0;
	margin-top: 6em;
}
.login-btn {
	font-family: $font_1;
	border: none;
	border-radius: .51em;
	height: 2.3em;
	background-color: $color_silver_chalice;
	color: $white;
	font-weight: 400;
	font-size: 1.3rem;
	cursor: pointer;
	letter-spacing: .1rem;
	padding: .4em;
	margin: 1em;
	&:hover {
		background-color: $color_mountain_mist;
	}
}
.logo-div {
	display: flex;
	justify-content: center;
	align-items: center;
	&:after {
		clear: both;
		display: block;
		content: '';
	}
	img {
		width: 4em;
		height: 4em;
		&:hover {
			animation: shake .5s ease-in-out;
		}
	}
	h1 {
		font-family: $font_1, $font_3;
		font-size: 4em;
		margin: .4em;
	}
}

//logo animations
@keyframes shake {
	0% {
		transform: translateX(0);
	}
	20% {
		transform: translateX(-.5em);
	}
	40% {
		transform: translateX(.2em);
	}
	60% {
		transform: translateX(-.5em);
	}
	80% {
		transform: translateX(.2em);
	}
	100% {
		transform: translateX(0);
	}
}

//mobile styles
@media screen and(min-width: 320px) and(max-width: 400px) {
	.logo-div {
		margin-top: 2em;
		display: block;
		h1 {
			margin: 0;
		}
	}
}