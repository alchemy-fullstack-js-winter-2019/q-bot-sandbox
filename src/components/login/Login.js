import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { subscribe, loginWithProvider } from '../../services/firebase';
import logo from '../../assets/logo.png';
import './Login.scss';

export default class Login extends Component {
  state = {
    provider: ''
  }

  componentDidUpdate() {
    this.subscribeWithLogin();
  }

  subscribeWithLogin() {
    this.unsubscribe && this.unsubscribe();
    this.unsubscribe = subscribe(user => {
      if(user) {
        this.props.history.replace('/questions');
      }
    }, () => {
      loginWithProvider(this.state.provider)
        .catch(console.log)
      })
  }

  componentDidMount() {
    this.subscribeWithLogin()
  }

  componentWillUnmount() {
    this.unsubscribe && this.unsubscribe();
  }

  handleClick = ({ target }) => {
    this.setState({ provider: target.value })
  }

  render() {
    return (
      <>
        <main>
          <div className={'logo-div'}>
            <img src={logo} alt="logo"/>
            <h1>Q BOT</h1>
          </div>
          <br/>
          <span>Sign in with:</span>
          <br/>
          <button className={'login-btn'}
            name='google'
            value='google'
            onClick={this.handleClick}
          >
            Google
          </button>
          <button className={'login-btn'}
            name='github'
            value='github'
            onClick={this.handleClick}
          >
            GitHub
          </button>
          <br/>
          <br/>
          <Link to="/about">About Us</Link>
        </main>
      </>
    );
  }
 
}
