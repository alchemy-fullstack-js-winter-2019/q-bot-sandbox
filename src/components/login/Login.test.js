import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Questions', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Login />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
