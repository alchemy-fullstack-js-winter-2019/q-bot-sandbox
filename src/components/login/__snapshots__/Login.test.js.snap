// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions matches a snapshot 1`] = `
<Fragment>
  <main>
    <div
      className="logo-div"
    >
      <img
        alt="logo"
        src="/path/image"
      />
      <h1>
        Q BOT
      </h1>
    </div>
    <br />
    <span>
      Sign in with:
    </span>
    <br />
    <button
      className="login-btn"
      name="google"
      onClick={[Function]}
      value="google"
    >
      Google
    </button>
    <button
      className="login-btn"
      name="github"
      onClick={[Function]}
      value="github"
    >
      GitHub
    </button>
    <br />
    <br />
    <Link
      to="/about"
    >
      About Us
    </Link>
  </main>
</Fragment>
`;
